src/main/webapp/app/components/util/short-answer-question-util.service.ts

```diff
@@ -0,0 +1,266 @@
+import { Injectable } from '@angular/core';
+import { ShortAnswerSolution } from '../../entities/short-answer-solution';
+import { ShortAnswerSpot } from '../../entities/short-answer-spot';
+import { ShortAnswerMapping } from '../../entities/short-answer-mapping';
+import { ShortAnswerQuestion} from '../../entities/short-answer-question';
+
+@Injectable({ providedIn: 'root' })
+export class ShortAnswerQuestionUtil {
+    constructor() {}
+
+    /**
+     * Get a sample solution for the given drag and drop question
+     *
+     * @param question {object} the drag and drop question we want to solve
+     * @param [mappings] {Array} (optional) the mappings we try to use in the sample solution (this may contain incorrect mappings - they will be filtered out)
+     * @return {Array} array of mappings that would solve this question (may be empty, if question is unsolvable)
+     */
+    solveSA(question: ShortAnswerQuestion, mappings: ShortAnswerMapping[]) {
```

Comment:
try to avoid (in the whole file) to abbreviate ShortAnswer with SA

Reply:
changed in branch sa-quiz-statistics